### PR TITLE
Don't serve config from kops-controller for APIServer nodes

### PR DESCRIPTION
**Pull request by hakman**
Closes https://github.com/kubernetes/kops/issues/14603

/cc @johngmyers @olemarkus

### CODE REVIEW

File: pkg/bootstrap/authenticate.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ type VerifyResult struct {
 
 	// InstanceGroupName is the name of the kops InstanceGroup this node is a member of.
 	InstanceGroupName string
+	// InstanceGroupRole is the role of the kops InstanceGroup this node is a member of.
+	InstanceGroupRole string
```
Comment (johngmyers) [inline pkg/bootstrap/authenticate.go:36]: Can't kops-controller's common code figure out the role of any given instance group name?

File: upup/pkg/fi/nodeup/nodetasks/service.go [hunk 2]
```diff
@@ -75,8 +74,12 @@ func (p *Service) GetDependencies(tasks map[string]fi.Task) []fi.Task {
 		switch v := v.(type) {
 		case *Package, *UpdatePackages, *UserTask, *GroupTask, *Chattr, *BindMount, *Archive, *Prefix, *UpdateEtcHostsTask:
 			deps = append(deps, v)
-		case *Service, *LoadImageTask, *PullImageTask, *IssueCert, *BootstrapClientTask, *KubeConfig:
+		case *Service, *PullImageTask, *IssueCert, *BootstrapClientTask, *KubeConfig:
```
Comment (johngmyers) [inline upup/pkg/fi/nodeup/nodetasks/service.go:77]: Should this be a separate PR?

Comment (hakman) [reply upup/pkg/fi/nodeup/nodetasks/service.go:77]: Yeah, was just playing around to see if it helps. Seems a good idea to keep and put in a separate PR.

Comment (hakman) [reply upup/pkg/fi/nodeup/nodetasks/service.go:77]: https://github.com/kubernetes/kops/pull/14613

**Review by johngmyers — APPROVED** [review]
Makes sense
